controllers/recipe.js

```
@@ -0,0 +1,147 @@
+    console.log(req.body)
+    const {
+        body: { recipeName, createdBy, cuisineType, estTimeOfPrep, ingredients, prepInstructions }
+    } = req
+    try {
+            if(!recipeName || !cuisineType || !estTimeOfPrep || !ingredients || !prepInstructions) {
+                throw new BadRequestError('Fields cannot be empty.')
+            } else {
+                console.log('we got here')
+                const newRecipe = {
+                    recipeName: recipeName,
+                    createdBy: createdBy,
+                    cuisineType: cuisineType,
+                    estTimeOfPrep: estTimeOfPrep,
+                    ingredients: ingredients,
+                    prepInstructions: prepInstructions
+                }
+                const recipes = await Recipe.create(newRecipe)
+                console.log(recipes)
+                res.status(200).json({recipes})
+            }
+        } catch (error) {
+            console.log(error)
+            if (error instanceof BadRequestError) {
+                res.status(500).json({ message: error.message });
+            }
+        }
+    }
+
+
+const editRecipe = async (req, res) => {
+    const id = req.params.id
+    const recipe = Recipe.findById(id)
+
+    const updatedData = { $set: {
+        recipeName: req.body.recipeName || recipe.recipeName,
+        cuisineType: req.body.cuisineType || recipe.cuisineType,
+        estTimeOfPrep: req.body.estTimeOfPrep || recipe.estTimeOfPrep,
+        ingredients: req.body.ingredients || recipe.ingredients,
+        prepInstructions: req.body.prepInstructions || recipe.prepInstructions
+    }}
+
+    try {
+        const updatedRecipe = await Recipe.findByIdAndUpdate({_id: id}, updatedData, { new: true, runValidators: true })
+        if (!updatedRecipe) {
+            throw new BadRequestError('Recipe could not be updated.')
+        } else {
+            res.status(200).json({updatedRecipe})
+        }
+    } catch (error) {
```

In the catch block, there are three scenarios to consider. If the error is of the type NotFoundError, you should return a 404 error along with an appropriate message. If the error is a ValidationError, you should return a 400 error and include the error message. For all other types of errors, return a 500 error.

```
catch (error) {
        if (error instanceof NotFoundError) {
            res.status(404).json({ message: error.message });
        } else if(error.name == "ValidationError") {
            res.status(400).json({message: error.message})
        } else {
            res.status(500).json({ message: 'Internal Server Error.' });
        }
    }
```